sklearn/utils/multiclass.py

```diff
@@ -301,21 +301,37 @@ def _check_partial_fit_first_call(clf, classes=None):
     set on ``clf``.
 
     """
-    if getattr(clf, 'classes_', None) is None and classes is None:
-        raise ValueError("classes must be passed on the first call "
-                         "to partial_fit.")
-
-    elif classes is not None:
-        if getattr(clf, 'classes_', None) is not None:
-            if not np.array_equal(clf.classes_, unique_labels(classes)):
-                raise ValueError(
-                    "`classes=%r` is not the same as on last call "
-                    "to partial_fit, was: %r" % (classes, clf.classes_))
+    if (getattr(clf, 'classes_', None) is None
+            and getattr(clf, 'classes', None) is None
+            and classes is None):
 
+        raise ValueError("classes must be passed on the first call "
+                         "to partial_fit or set during initialization"
+                         "if the class has a classes parameter.")
+
+    if (classes is not None
+            and getattr(clf, 'classes', None) is not None):
+        if not np.array_equal(clf.classes, unique_labels(classes)):
```

Comment:
though i've removed one case because it was redundant, but unique_labels is not applied on the classes passed at __init__, but on the classes passed in partial fit. The condition of requiring sorted input was not levied there earlier so I guess we should not change that behavior.